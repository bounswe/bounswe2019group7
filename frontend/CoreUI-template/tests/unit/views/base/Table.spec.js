import Vue from 'vue'
import { mount, shallowMount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue'
import Table from '@/views/base/Table'

Vue.use(BootstrapVue)

describe('Table.vue', () => {
  it('has a name', () => {
    expect(Table.name).toMatch('c-table')
  })
  it('has a created hook', () => {
    expect(typeof Table.data).toMatch('function')
  })
  it('sets the correct default data', () => {
    expect(typeof Table.data).toMatch('function')
    const defaultData = Table.data()
    expect(defaultData.currentPage).toBe(1)
  })
  it('is Vue instance', () => {
    const wrapper = mount(Table)
    expect(wrapper.isVueInstance()).toBe(true)
  })
  it('is Table', () => {
    const wrapper = mount(Table)
    expect(wrapper.is(Table)).toBe(true)
  })
  it('should render correct content', () => {
    const wrapper = mount(Table)
    expect(wrapper.find('div.card-header > div').text()).toMatch('Table')
  })
  test('renders correctly', () => {
    const wrapper = shallowMount(Table)
    expect(wrapper.element).toMatchSnapshot()
  })
})
