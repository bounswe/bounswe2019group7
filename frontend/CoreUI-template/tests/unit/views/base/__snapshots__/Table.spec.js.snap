// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table.vue renders correctly 1`] = `
<b-card-stub
  bodytag="div"
  footertag="div"
  headertag="div"
  subtitle=""
  subtitletag="h6"
  subtitletextvariant="muted"
  tag="div"
  title=""
  titletag="h4"
>
  <div
    slot="header"
  >
    Table
  </div>
   
  <b-table-stub
    apiurl=""
    currentpage="1"
    emptyfilteredtext="There are no records matching your request"
    emptytext="There are no records to show"
    fields=""
    items=""
    labelsortasc="Click to sort Ascending"
    labelsortclear="Click to clear sorting"
    labelsortdesc="Click to sort Descending"
    perpage="5"
    responsive="sm"
    selectedvariant="primary"
    selectmode="multi"
    sortby=""
    sortdirection="asc"
    value=""
  />
   
  <nav>
    <b-pagination-stub
      align="left"
      arialabel="Pagination"
      ellipsistext="…"
      firsttext="«"
      hidegotoendbuttons="true"
      labelfirstpage="Go to first page"
      labellastpage="Go to last page"
      labelnextpage="Go to next page"
      labelpage="Go to page"
      labelprevpage="Go to previous page"
      lasttext="»"
      limit="5"
      nexttext="Next"
      perpage="5"
      prevtext="Prev"
      size="md"
      totalrows="0"
      value="1"
    />
  </nav>
</b-card-stub>
`;
